src/Microsoft.Diagnostics.EventFlow.Inputs.DiagnosticSource/DiagnosticSourceInput.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Diagnostics.EventFlow.Configuration;
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs.DiagnosticSource
+{
+    public sealed class DiagnosticSourceInput : IObservable<EventData>, IDisposable
+    {
+        private readonly DiagnosticListenerObserver _listenerObserver;
+        private readonly IDisposable _listenerSubscription;
+        private readonly EventFlowSubject<EventData> _subject;
+
+        public DiagnosticSourceInput(IReadOnlyCollection<DiagnosticSourceConfiguration> sources, IHealthReporter healthReporter)
+        {
+            if (sources == null)
```

Comment:
(minor) We use Validation package in other parts of EventFlow, so you can be a bit less verbose and say

```csharp
Requires.NotNull(sources, nameof(sources));
```
E.g. https://github.com/Azure/diagnostics-eventflow/blob/master/src/Microsoft.Diagnostics.EventFlow.Inputs.Etw/EtwInput.cs#L31

(just a suggestion)

Reply:
I'm not really a fan of such libraries. With the changes to the last few versions of the C# language, like `nameof` and being able to `throw` from `??` and `?:` they aren't as necessary, barely saving any code, and add extra dependencies and potentially supply chain attack vectors.

The gold partner ISV where I work is quite strict on which packages can be used in our software, and the dependency on the Validation package is the kind of reason they may reject Microsoft.Diagnostics.EventFlow.

Although there is already a dependency on Validation, I didn't want to make it any worse.

Reply:
OK, it is not a big deal

For the sake of your partner ISV: I also do not like unnecessary dependencies, and as one can see, EventFlow has not many. The Validation library has a minimalist design that I very much like. In addition to cutting down the amount of code, adopting Validation library improves readability of the code--it makes argument checks and assumptions very explicit. I also happen to know the author; he is a fellow Microsoft employee and a very conscientious developer. I have been using Validation for various project for many years and my concern about this library having negative impact on EventFlow is zero.